deuce/model/vault.py

```
@@ -107,13 +109,20 @@ def put_async_block(self, block_ids, blockdatas):
                                                                   storage_ids,
                                                                   blockdatas,
                                                                   block_sizes):
+                logger.info('Project {0}, Vault {1}: Associating metadata '
+                            'block {2} with storage block {3}'
+                            .format(deuce.context.project_id,
+                                    self.id,
+                                    block_id,
+                                    storageid))
                 deuce.metadata_driver.register_block(
                     self.id,
                     block_id,
                     storageid,
                     block_size)
+                retblocks.append((block_id, storageid))
```

I would say let it fail as a whole...  Fine grained error suggests fine grained error handling, which I don't think a client would like to try.  And actually, if you just return 503, the current c++ client will retry the whole bundle, problem solved.